segmenter_test.go

```diff
@@ -72,3 +73,62 @@ func TestPathSegmenterEdgeCases(t *testing.T) {
 		}
 	}
 }
+
+type testDepth struct {
+	cases map[string]bool
+	arm   bool
+}
+
+func (t *testDepth) testPathSegmenterDot(path string, start int) (segment string, next int) {
+	if len(path) == 0 || start < 0 || start > len(path)-1 {
+		return "", -1
+	}
+	end := strings.IndexRune(path[start+1:], '.')
+	if end == -1 {
+		return path[start:], -1
+	}
+	if t.arm {
+		t.cases[path] = true
+	}
+	return path[start : start+end+1], start + end + 1
+}
+
+func TestCustomPathSegmenter(t *testing.T) {
+	depthTest := testDepth{
+		cases: map[string]bool{
```

Comment:
I'd comment this `// key -> visited` and move it to be after the `tie` is created to be more clear